### PR TITLE
Always push down in filter to Kudu

### PR DESCRIPTION
**Pull request by twdsilva**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java [hunk 1]
```diff
@@ -162,6 +162,23 @@ public void testProjectionWithFilter() throws Exception {
     }
   }
 
+  @Test
+  public void testInFilterAlwaysPushedDown() throws Exception {
+    try (Connection conn = DriverManager.getConnection(JDBC_URL)) {
+      String sqlFormat = "SELECT sid FROM \"ReportCenter.DeliveredMessages\" WHERE "
+          + "account_sid = '%s' AND mcc IN ('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20')";
+      String sql = String.format(sqlFormat, JDBCQueryIT.ACCOUNT_SID);
+      String expectedPlan = "KuduToEnumerableRel\n" + "  KuduProjectRel(SID=[$2])\n"
```
Comment (minakshikorad) [inline adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java:171]: Does this test result in hash join without the change in `withInSubQueryThreshold`  KuduSchema.java?

Comment (twdsilva) [reply adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java:171]: Yes if the IN list contains at least 20 elements
https://github.com/apache/calcite/blob/f7bf867ba96578e7cc40e2aac1adf47015d06913/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L6555

Comment (sdreynolds) [reply adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java:171]: Looking at that code, it seems it is related to turning a SQL Node into a RelNode.

What happens if you start with a Logical RelNode plan? I don't think this configuration is referenced when that is used.

Comment (twdsilva) [reply adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java:171]: Thats a good point let me figure out how to hook this if we use RelBuilder.

Comment (twdsilva) [reply adapter/src/test/java/com/twilio/kudu/sql/JDBCQueryIT.java:171]: This isnt a issue if you use the RelBuilder api, the IN list isn't converted to a hashjoin.

**Comment by twdsilva** [comment]
Not sure why the tests are failing because of this now
```
/tmp/kudu-binary-jar2059560088047511673/kudu-binary-1.16.0-linux-x86_64/bin/kudu: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory
```

**Review by rbhanot — APPROVED** [review]

**Pull request merged**